pwa/src/components/common/layout.tsx

```diff
@@ -22,9 +23,26 @@ import favicon from "../../images/conduction_logo_blauw.svg";
 export default function Layout({ children, title = "", subtext = "" }) {
   const [API, setAPI] = React.useState<APIService>(null)
 
+  const parseJwt = (token) => {
+    try {
+      return JSON.parse(atob(token.split(".")[1]));
+    } catch (e) {
+      return null;
+    }
+  };
+  const decodedJwt = parseJwt(sessionStorage.getItem('jwt'));
+
   React.useEffect(() => {
+    if (decodedJwt === null) {
+      logout()
+      navigate("/");
+    } else if (decodedJwt.exp * 1000 < Date.now()) {
```

Comment:
I would stay away from `else if` statements as much as possible. In my opinion, this could just be another if statement, like so:

`if (Date.now() >= decodedJwt?.exp * 1000)` Note: see the ? after decodedJwt, this firstly does the check if `decodedJwt` is available at all.